js/pro/gate.js

```diff
@@ -741,23 +741,48 @@ module.exports = class gate extends gateRest {
     }
 
     async watchOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name gate#watchOrders
+         * @description watches information on multiple orders made by the user
+         * @see https://www.gate.io/docs/developers/apiv4/ws/en/#orders-channel
+         * @see https://www.gate.io/docs/developers/futures/ws/en/#orders-api
+         * @see https://www.gate.io/docs/developers/options/ws/en/#orders-channel
+         * @param {string|undefined} symbol unified market symbol of the market orders were made in
+         * @param {int|undefined} since the earliest time in ms to fetch orders for
+         * @param {int|undefined} limit the maximum number of  orde structures to retrieve
+         * @param {object} params extra parameters specific to the gate api endpoint
+         * @param {string} params.type spot, margin, swap, future, or option. Required if listening to all symbols.
+         * @param {boolean} params.isInverse if future, listen to inverse or linear contracts
+         * @returns {[object]} a list of [order structures]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure
+         */
+        let type = undefined;
+        let market = undefined;
         if (symbol === undefined) {
-            throw new ArgumentsRequired (this.id + ' watchOrders requires a symbol argument');
-        }
-        await this.loadMarkets ();
-        const market = this.market (symbol);
-        symbol = market['symbol'];
-        let type = 'spot';
-        if (market['future'] || market['swap']) {
-            type = 'futures';
-        } else if (market['option']) {
-            type = 'options';
+            const [ defaultType ] = this.handleMarketTypeAndParams ('watchOrders', undefined, params);
+            type = defaultType;
+        } else {
+            await this.loadMarkets (false, params);
+            market = this.market (symbol);
+            symbol = market['symbol'];
+            type = market['type'];
         }
-        const method = type + '.orders';
+        const typeId = this.getSupportedMapping (type, {
+            'spot': 'spot',
+            'margin': 'spot',
+            'future': 'futures',
+            'swap': 'futures',
+            'option': 'options',
+        });
+        const method = typeId + '.orders';
         let messageHash = method;
-        messageHash = method + ':' + market['id'];
-        const url = this.getUrlByMarketType (market['type'], market['inverse']);
-        const payload = [ market['id'] ];
+        let payload = [ '!all' ];
+        if (symbol !== undefined) {
+            messageHash = method + ':' + market['id'];
+            payload = [ market['id'] ];
+        }
+        const isInverse = this.safeValue (params, 'isInverse', false);
```

Comment:
`isInverse` inside `params` is not the unified way to get it, we should follow this order:
-> check the symbol's subtype if provided
-> check params for `subType` value
-> check defaultSubType option

(we could probably create `handleSubTypeAndParams` to unify these checks )

Reply:
@pcriadoperez can you check this comment? ⬆️

Reply:
Fixed, I saw we actually already have a `handleSubTypeAndParams`

Reply:
@pcriadoperez nice, will take a look now